### ISSUE TITLE
SDL not found

### ISSUE THREAD

**Issue by Danfun64** [ata4/angrylion-rdp-plus#28]
I'm sure that both SDL and SDL2 are installed on my MSYS2/MinGW-W64 setup... but for some reason, when I try to compile this, it doesn't see it. http://paste.debian.net/989661/

**Comment by loganmc10** [comment]
If you do get it to compile you'll find that it performs terribly using MinGW. The c++11 threading implementation in MinGW performs awfully. You'll have much better performance using Visual Studio

But more to this point, you're probably missing pkg-config. I can't remember what the full package name is in MinGW, but it contains either pkg-config or pkgconfig.

@ata4 it would be nice if retrace wasn't built automatically when trying to build the mupen64plus package. SDL isn't actually even a requirement for what @Danfun64 is trying to build

**Comment by Danfun64** [comment]
It has pkg-config...but I don't know what commands are necessary. "Must specify package names on the command line"

**Comment by loganmc10** [comment]
I thought cmake used pkgconfig to determine your SDL version but I guess not, I'm not sure why it's not working, but like I mentioned, the root issue is that it's trying to build retrace, which you don't need, the actual mupen64plus plugin doesn't depend on SDL

**Comment by SigmaVirus** [comment]
I know this is pretty irrelevant, but if you have a hard time building and just want to try out the plugin for Project64 I'll share a build from 2017-10-07 for you ready to go.

[angrylion-plus (17-10-07).zip](https://github.com/ata4/angrylion-rdp-plus/files/1367129/angrylion-plus.17-10-07.zip)

**Comment by ata4** [comment]
SDL2 is a soft requirement anyway. It's currently being used by the retracer only, which again is used mostly for testing and debugging without having to use an emulator. I think I should disable it on default to make building easier.

**Comment by Danfun64** [comment]
At the very least, how does one fix things so that SDL2 works properly for the retracer?

**Comment by ata4** [comment]
It's possible that you may have to change the include path from `#include <SDL.h>` to `#include <SDL2/SDL.h` in screen.c and retrace.c, or disable the `#ifdef` there. On my Linux machine, it only compiles when using just `SDL.h`, so maybe it's the same in MinGW.

**Comment by Danfun64** [comment]
Does this project use SDL 1.2 at all? I removed the <SDL.h> lines leaving in only the <SDL2/SDL.h> lines and got this: http://paste.debian.net/992183/

**Comment by ata4** [comment]
It uses SDL 2.0. That WinMain linking error seems familiar, I think I got this error in the past on SDL applications several times. I'll try to compile it with MinGW myself and fix the CMake file, since I want to investigate that performance issue, anyway.

**Comment by loganmc10** [comment]
I think the WinMain error is because you need to link the "SDL2" library as well as "SDL2main", they are separated (like ```-lSDL2 -lSDL2main``` for example in Linux)

**Comment by ata4** [comment]
I made the retracer and its SDL2 requirement optional, which can be re-enabled with `-DRETRACER=on`. This should fix all issues for everyone who just wants to compile the plugin.

I also made some tests with MinGW and it looks like it requires `include_directories(${SDL2_INCLUDE_DIR})` in order to find the SDL2 headers. I'm not sure if it's a CMake bug or a configuration error, so I better won't add that line as a hack.
